src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -1066,6 +1068,7 @@ def load_arguments(self, _):
         # virtual machines
         c.argument("vm_sizes", is_preview=True)
         c.argument("enable_imds_restriction", action="store_true", is_preview=True)
+        c.argument("managed_system_pool", action="store_true", is_preview=True)
```

Comment:
It seems to me that `--managed-system-pool` isn't intended to function as a boolean switch. Perhaps `--enable-managed-system-pool` would be a more appropriate alternative.